.buildbot.sh

```diff
@@ -2,12 +2,25 @@
 #
 # Build script for continuous integration.
 
-./x.py clean  # We don't clone afresh to save time and bandwidth.
-git clean -dffx # If upstream removes a submodule, remove the files from disk.
+set -e
 
 # Ensure the build fails if it uses excessive amounts of memory.
 ulimit -d $((1024 * 1024 * 8)) # 8 GiB
 
 # Note that the gdb must be Python enabled.
 /usr/bin/time -v env PATH=/opt/gdb-8.2/bin:${PATH} \
     RUST_BACKTRACE=1 ./x.py test --config .buildbot.toml
+
+# Archive the build and put it in /opt
+GIT_HASH=`git rev-parse --short HEAD`
+TARBALL_TOPDIR=ykrustc-stage2
+TARBALL_NAME=ykrustc-stage2-${GIT_HASH}.tar.bz2
+SNAP_DIR=/opt/ykrustc-bin-snapshots
+
+cd build/x86_64-unknown-linux-gnu
+ln -sf stage2 ${TARBALL_TOPDIR}
+git show -s HEAD > ${TARBALL_TOPDIR}/VERSION
+tar hjcvf ${TARBALL_NAME} ${TARBALL_TOPDIR}
+chmod 775 ${TARBALL_NAME}
+rm -f ${SNAP_DIR}/ykrustc-stage2-*.tar.bz2
```

Comment:
Not sure I understand why this line is here.

Reply:
It's removing the old binary snapshot, if it exists.

Reply:
Ah, shouldn't it do `rm -f  ${SNAP_DIR}/${TARBALL_NAME}`?

Reply:
No, because we don't know the filename of the previous build.

Note the short git hash in the filename.

As I alluded to in the description, I think in the longer term we want some kind of rolling back-log of the most recent (say) 5 builds, but let's not complicate things unless we really need to ;)

Reply:
Why would there be an old build lurking around? I'm generally wary about `*` and `rm` unless we really can't do anything about it!

If you want to make sure the current tarball is always deleted you could do something like `trap 'rm -f $blah' 1 2 3 13 15`?

Reply:
I think you misunderstand.

Suppose we've already done a build in the past. `/opt/ykrustc-bin-snapshots` will already contain a tarball, say `ykrustc-stage2-111111.tar.bz2`. Then the next build will make a new tarball with a different name, say `ykrustc-stage2-222222.tar.bz2`.

At then end of the second build, we remove the old tarball before moving in the new one. This is for no other reason than to not fill the disk.

Reply:
Ah. Why are we complicating this? Shouldn't we just write to `/opt/ykrustc-bin-snapshots/latest.tar.bz2` for simplicity, at least at first?

Reply:
We can do, if you are content with only the file's date stamp as an identifier.

Of course you can untar it and look athe the VERSION file.

Reply:
I'd prefer to start simple and build the extra complexity later if we find we need it.